### PR TITLE
Refactor CLI using dtyper

### PR DESCRIPTION
**Pull request by disrupted**
Typer doesn't allow calling commands from outside its context. `dtyper` acts as a slim wrapper to patch this drawback, making it possible to import the commands from an external Python module. This is the first step towards providing a developer API that allows to hook into the kpops operations.

### CODE REVIEW

File: kpops/cli/main.py [hunk 1]
```diff
@@ -185,7 +190,7 @@ def create_pipeline_config(
     return pipeline_config
 
 
-@app.command(
+@app.command(  # pyright: ignore[reportGeneralTypeIssues]
```
Comment (disrupted) [inline kpops/cli/main.py:193]: opened an issue https://github.com/rec/dtyper/issues/8

Comment (irux) [reply kpops/cli/main.py:193]: I think it would be better to put the link on the code too :)

File: tests/pipeline/test_pipeline.py [hunk 2]
```diff
@@ -5,23 +5,33 @@
...
+        assert len(pipeline) == 3
+
     def test_load_pipeline(self, snapshot: SnapshotTest):
         result = runner.invoke(
             app,
             [
                 "generate",
                 "--pipeline-base-dir",
-                PIPELINE_BASE_DIR,
+                str(PIPELINE_BASE_DIR_PATH),
```
Comment (irux) [inline tests/pipeline/test_pipeline.py:34]: Any reason to move the str to the reference and remove it from the definition ?

Comment (disrupted) [reply tests/pipeline/test_pipeline.py:34]: for the new test which needs a `Path` object. besides, it looks a little more uniform with the rest of the arguments

File: tests/pipeline/test_pipeline.py [hunk 3]
```diff
@@ -5,23 +5,33 @@
 from snapshottest.module import SnapshotTest
 from typer.testing import CliRunner
 
+import kpops
 from kpops.cli.main import app
 from kpops.pipeline_generator.pipeline import ParsingException
 
 runner = CliRunner()
 
 RESOURCE_PATH = Path(__file__).parent / "resources"
```
Comment (irux) [inline tests/pipeline/test_pipeline.py:14]: os.join or use Path for joining it already ?

**Comment by raminqaf** [comment]
Closing PR because this can now be achieved through https://github.com/bakdata/kpops/pull/306 and doesn't need to be natively built in KPOps

File: tests/pipeline/test_pipeline.py (hunk 3 above)
Comment (disrupted) [reply tests/pipeline/test_pipeline.py:14]: what do you mean? the `/` operator works the same as `Path(...).join(...)`

**Review by raminqaf — APPROVED** [review]

**Review by irux — APPROVED** [review]

**Pull request merged**